src/pybind/mgr/dashboard/HACKING.rst

```diff
@@ -3,6 +3,142 @@ Ceph Dashboard Developer Documentation
 
 .. contents:: Table of Contents
 
+Preliminary Steps
+-----------------
+
+The following documentation chapters expect a running Ceph cluster and at
+least a running `dashboard` manager module (with few exceptions). This
+chapter gives an introduction on how to set up such a system for development,
+without the need to set up a full blown production environment. All in this
+chapter introduced options are based on a so called `vstart` environment. 
+
+.. note::
+
+  Every `vstart` environment needs Ceph `to be compiled`_ from its Github
+  repository, though Docker environments simplify that step by providing a
+  shell script that contains those instructions.
+
+.. _to be compiled: https://docs.ceph.com/docs/master/install/build-ceph/
+
+vstart
+~~~~~~
+
+"vstart" is actually a shell script in the `src/` directory of the Ceph
+repository (``src/vstart.sh``). It is used to start a single node Ceph
+cluster on the machine where it is executed. Several required and some
+optional Ceph internal services are started automatically when it is used to
+start a Ceph cluster. `vstart` is the basis for the three most commonly used
+development environments in Ceph Dashboard.
+
+Host vs Docker
+~~~~~~~~~~~~~~
+
+This document introduces you to three different development environments, all
+based on vstart. Those are:
+
+- vstart running on your host system
+
+- vstart running in a Docker environment
+
+  * ceph-dev-docker_
+  * ceph-dev_
+
+.. note::
+
+  Independently of which of these environments you will choose, you need to
+  compile Ceph in that environment. If you compiled Ceph on your host system,
+  you would have to recompile it on Docker to be able to switch to a Docker
+  based solution. The same is true vice versa. If you previously used a
+  Docker development environment and compiled Ceph there and you now want to
+  switch to your host system, you will also need to recompile Ceph (or
+  compile Ceph using another separate repository).
+
+Development environment on your host system
+...........................................
+
+- No need to learn or have experience with Docker, jump in right away
+- Limited amount of scripts to support automation (like Ceph compilation)
+- No pre-configured easy-to-start services (Prometheus, Grafana, etc)
+- Limited amount of host operating systems supported, depending on which
+  Ceph version is supposed to be used
+
+Development environment using Docker
```

Comment:
```suggestion
Development environments based on Docker/Podman
```

Reply:
Unless someone assures me that `Podman` really works on either environment, I'd prefer not to mention it. From my experience, the switch from `Docker` to `Podman` is nearly seamless, but not so from `docker-compose` to `podman-compose`. I do not want people to think they could use either of them, but then failing miserably by trying to use `podman-compose`, which I honestly expect not to function with our averagely complex `docker-compose.yml` files.